### PR TITLE
doc: Add linting hints and docs

### PR DESCRIPTION
**Pull request by cknight**
This PR contributes to #159 by adding linting hints and docs to:
* no_irregular_whitespace
* no_misued_new

### CODE REVIEW

File: src/rules/no_irregular_whitespace.rs [hunk 1]
```diff
@@ -175,149 +219,174 @@ mod tests {
...
         }
       ],
       "var any \u{2009} = 'thing';": [
         {
           col: 8,
           message: NoIrregularWhitespaceMessage::NotAllowed,
+          hint: HINT,
         }
-      ],
+        ],
```
Comment (magurotuna) [inline src/rules/no_irregular_whitespace.rs:315]:
```suggestion
      ],
```

Looks like more indent than it should have

File: src/rules/no_misused_new.rs [hunk 2]
```diff
@@ -28,6 +28,35 @@ impl LintRule for NoMisusedNew {
...
+```
+
+### Valid:
+```typescript
+class C {
+  constructor() {}
+}
+interface I {
+  new (): C;
+}
```
Comment (magurotuna) [inline src/rules/no_misused_new.rs:56]:
```suggestion
class C {
  constructor() {}
}

interface I {
  new (): C;
}
```


a very minor nit, but it'd be nice to have a blank line between `C` and `I` as the `Invalid` example has 👍

**Review by magurotuna — CHANGES_REQUESTED** [review]

**Review by magurotuna — APPROVED** [review]
LGTM 👍  nice doc!

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks you @cknight!

**Pull request merged**